src/main/java/jenkins/plugins/slack/workflow/SlackSendStep.java

```diff
@@ -0,0 +1,106 @@
+package jenkins.plugins.slack.workflow;
+
+import hudson.AbortException;
+import hudson.Extension;
+import hudson.model.Run;
+import hudson.model.TaskListener;
+import jenkins.model.Jenkins;
+import jenkins.plugins.slack.SlackNotifier;
+import jenkins.plugins.slack.SlackService;
+import jenkins.plugins.slack.StandardSlackService;
+import jenkins.plugins.slack.Messages;
+import org.jenkinsci.plugins.workflow.steps.AbstractStepDescriptorImpl;
+import org.jenkinsci.plugins.workflow.steps.AbstractStepImpl;
+import org.jenkinsci.plugins.workflow.steps.AbstractSynchronousNonBlockingStepExecution;
+import org.jenkinsci.plugins.workflow.steps.StepContextParameter;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.DataBoundSetter;
+
+import javax.annotation.Nonnull;
+import javax.inject.Inject;
+import java.util.logging.Logger;
+
+/**
+ * Workflow step to send a Slack channel notification.
+ */
+public class SlackSendStep extends AbstractStepImpl {
+
+	private static final Logger logger = Logger.getLogger(SlackSendStep.class.getName());
+
+	public final String message;
+
+	@DataBoundSetter
+	public String color;
```

Comment:
+1, non-final public fields are actually serious design :bug:s in the modern world